### DISCUSSION TITLE
Limit number of events shown per category

### DISCUSSION THREAD

**Discussion by StickyPixels** [liedekef/events-made-easy#202] [General]
What I'm trying to achieve is to display a chronological list of events, showing only the most recent event in each category. But whatever I try, all events show up in the list. Apparently it is not possible to limit the number of events per category?

This is the code I'm using, first to call the list of events:
`[eme_events scope=all order=DESC template_id=3 limit=0]`
Next I try to limit the number of events per category in template 3:
`[eme_events category=#_EVENTCATEGORYIDS scope=past template_id=9 limit=1]`
But all events in each category are displayed. Apparently the limit=1 part is ignored for categories, am I correct? Or am I completely missing something?

I'm not sure if what I'm trying to do is possible, and it's a very niche question, but I thought I'd try anyway. I have figured out a workaround but I'm trying to keep it as simple as possible for the end user.

**Comment by liedekef** [comment]
your first eme_events call will show all events already (scope=all and limit=0)
Currently it is not possible like this (looping over all existing categories). The eme_for shortcode could serve for this purpose though:
```
[eme_for list="3,4,5" sep=","]
 [eme_events category=#_LOOPVALUE scope=past template_id=9 limit=1]
[/eme_for]
```
Unfortunately currently there is no placeholder that replaces all category ids in one go in eme_for though ... some simple coding could solve this :-)

**Reply by liedekef** [reply]
This change https://github.com/liedekef/events-made-easy/commit/aa17575b1e558ceef98b9ae89be2eca57583bc87 , followed by https://github.com/liedekef/events-made-easy/commit/c8269ce26e405e112b3d62876358727d230e646b should allow
```
[eme_for list="#_ALLCATEGORYIDS"]
 [eme_events category=#_LOOPVALUE scope=past template_id=9 limit=1]
[/eme_for]
```

**Comment by StickyPixels** [comment]
Huge thanks for your instant reply and modifications. I changed the plugin files and pasted your suggested code in a test page, but I get a fatal error:
`Fatal error: Uncaught ArgumentCountError: Too few arguments to function eme_get_category_ids(), 0 passed in /_PATH_/wp-content/plugins/events-made-easy/eme-events.php on line 2238 and exactly 1 expected in /_PATH_/wp-content/plugins/events-made-easy/eme-categories.php:447 Stack trace: #0 /_PATH_/wp-content/plugins/events-made-easy/eme-events.php(2238): eme_get_category_ids() #1 /_PATH_/wp-content/plugins/events-made-easy/eme-functions.php(672): eme_replace_generic_placeholders('#_ALLCATEGORYID...') #2 /_PATH_/wp-includes/shortcodes.php(355): eme_for_shortcode(Array, '<br />\n [eme_ev...', 'eme_for') #3 [internal function]: do_shortcode_tag(Array) #4 /_PATH_/wp-includes/shortcodes.php(227): preg_replace_callback('/\\[(\\[?)(eme_ev...', 'do_shortcode_ta...', '[eme_for list="...') #5 /_PATH_/wp-includes/class-wp-hook.php(308): do_shortcode('[eme_for list="...') #6 /_PATH_/wp-includes/plugin.php(205): WP_Hook->apply_filters('[eme_for list="...', Array) #7 /_PATH_/wp-includes/post-template.php(255): apply_filters('the_content', '[eme_for list="...') #8 /_PATH_/wp-content/themes/Mariken/page.php(24): the_content() #9 /_PATH_/wp-includes/template-loader.php(106): include('/Users/username/Site...') #10 /_PATH_/wp-blog-header.php(19): require_once('/Users/username/Site...') #11 /_PATH_/index.php(17): require('/Users/username/Site...') #12 {main} thrown in /_PATH_/wp-content/plugins/events-made-easy/eme-categories.php on line 447`
Am I missing something of using the shortcode in the wrong place?

**Reply by liedekef** [reply]
This should fix that: https://github.com/liedekef/events-made-easy/commit/8c313ac6bfdbd6705f3dacd21c8782d83f693475

**Comment by StickyPixels** [comment]
You are amazing, so quick and helpful. Looks like it works exactly as intended! :))
Many thanks!

**Comment by StickyPixels** [comment]
Hi Franky,
Sorry to bother you again but there seems to be a little glitch in the code when I try to exclude multiple categories from the list. I now have this code:
`[eme_for list="#_ALLCATEGORYIDS"] [eme_events category=#_LOOPVALUE notcategory=24 scope=all template_id=9 limit=1] [/eme_for]`
This works fine, category 24 is excluded from the list.
But as soon as I try to exclude multiple categories it doesn't work, and no category is hidden:
`[eme_for list="#_ALLCATEGORYIDS"] [eme_events category=#_LOOPVALUE notcategory=24,25,26 scope=all template_id=9 limit=1] [/eme_for]`
Should this be possible, or is not possible to hide multiple categories?

**Reply by liedekef** [reply]
I thinkg best is to do: notcategory=24+25+26 (maybe even quotes around it) to exclude cat 24 AND 25 AND 26
Otherwise it will be read as "show events not in category 24 OR not in category 25 OR not in category 26" ==> "notcategory" and the "OR" will lead to unexpected results :-)

**Comment by StickyPixels** [comment]
Alright, that seems to work! :)) Thanks again!
